python/dgl/nn/pytorch/conv/edgegraphconv.py

```diff
@@ -294,20 +248,15 @@ def forward(self, graph, feat, weight=None, edge_weight=None):
             else:
                 weight = self.weight
 
-            if self._in_feats > self._out_feats:
-                # mult W first to reduce the feature size for aggregation.
-                if weight is not None:
-                    feat_src = th.matmul(feat_src, weight)
-                graph.srcdata['h'] = feat_src
-                graph.update_all(aggregate_fn, fn.sum(msg='m', out='h'))
-                rst = graph.dstdata['h']
-            else:
-                # aggregate first then mult W
-                graph.srcdata['h'] = feat_src
-                graph.update_all(aggregate_fn, fn.sum(msg='m', out='h'))
-                rst = graph.dstdata['h']
+            single_channel_rsts = []
```

Comment:
because of the way EGNN works, I think we might require that graph is fully connected and the edge feature matrix is (N (N + 1) / 2, P) or (N (N - 1) / 2, P) depending on whether we have self edges. In that case, I think we can do a simple matmul operation rather than a for loop. What are your thoughts?

Reply:
For a large graph, requiring that it be fully connected might be too strict -- why not let the user decide when it is appropriate to use this layer type? 

Pros of this current implementation (for loop):
more efficient 
matches the representation currently used for edge features 
Pros for matrix mul:
simpler to understand (for non-users of DGL)
avoids for loop

My ideal scenario would be to avoid this for loop by finding the correct DGL built-in to use.